// rete-angular-render-plugin/esm2015/socket/socket.component.js
import * as tslib_1 from "tslib";
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import { Input as ReteInput } from 'rete';
let SocketComponent = class SocketComponent {
    get type() {
        return this.io instanceof ReteInput ? 'input' : 'output';
    }
};
tslib_1.__decorate([
    Input()
], SocketComponent.prototype, "socket", void 0);
tslib_1.__decorate([
    Input()
], SocketComponent.prototype, "io", void 0);
SocketComponent = tslib_1.__decorate([
    Component({
        selector: 'rete-socket',
        template: `<div *ngIf="socket" class="socket" [ngClass]="[type, socket.name]" [title]="socket.name"></div>`,
        changeDetection: ChangeDetectionStrategy.OnPush,
        styles: [":host{display:inline-block}.socket{display:inline-block;cursor:pointer;border:1px solid #fff;border-radius:12px;width:24px;height:24px;margin:6px;vertical-align:middle;background:#96b38a;z-index:2;box-sizing:border-box}.socket:hover{border-width:4px}.socket.multiple{border-color:#ff0}.socket.output{margin-right:-12px}.socket.input{margin-left:-12px}"]
    })
], SocketComponent);
export { SocketComponent };
