// rete-angular-render-plugin/esm2015/custom.component.js
import * as tslib_1 from "tslib";
import { Component, Input, OnInit, Injector, ComponentFactoryResolver, ViewContainerRef, Type, ChangeDetectionStrategy } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
let CustomComponent = class CustomComponent {
    constructor(vcr, injector, factoryResolver) {
        this.vcr = vcr;
        this.injector = injector;
        this.factoryResolver = factoryResolver;
    }
    ngOnInit() {
        const factory = this.factoryResolver.resolveComponentFactory(this.component);
        const componentRef = factory.create(this.injector);
        const { props } = this;
        for (let key in props) {
            Object.defineProperty(componentRef.instance, key, {
                get() { return props[key]; },
                set(val) { props[key] = val; }
            });
        }
        this.vcr.insert(componentRef.hostView);
    }
    ngOnDestroy() {
        this.vcr.detach(0);
    }
};
CustomComponent.ɵfac = function CustomComponent_Factory(t) { return new (t || CustomComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ViewContainerRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.Injector), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ComponentFactoryResolver)); };
CustomComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CustomComponent, selectors: [["ng-component"]], inputs: { component: "component", props: "props" }, decls: 0, vars: 0, template: function CustomComponent_Template(rf, ctx) { }, encapsulation: 2, changeDetection: 0 });
CustomComponent.ctorParameters = () => [
    { type: ViewContainerRef },
    { type: Injector },
    { type: ComponentFactoryResolver }
];
tslib_1.__decorate([
    Input()
], CustomComponent.prototype, "component", void 0);
tslib_1.__decorate([
    Input()
], CustomComponent.prototype, "props", void 0);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CustomComponent, [{
        type: Component,
        args: [{
                template: '',
                changeDetection: ChangeDetectionStrategy.OnPush
            }]
    }], function () { return [{ type: ɵngcc0.ViewContainerRef }, { type: ɵngcc0.Injector }, { type: ɵngcc0.ComponentFactoryResolver }]; }, { component: [{
            type: Input
        }], props: [{
            type: Input
        }] }); })();
export { CustomComponent };
