// rete-angular-render-plugin/esm2015/socket.directive.js
import * as tslib_1 from "tslib";
import { Directive, Input, ElementRef, OnInit } from '@angular/core';
import { Input as ReteInput } from 'rete';
import { NodeService } from './node.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './node.service';
let SocketDirective = class SocketDirective {
    constructor(el, service) {
        this.el = el;
        this.service = service;
    }
    get type() {
        return this.io instanceof ReteInput ? 'input' : 'output';
    }
    ngOnInit() {
        this.service.bindSocket(this.el.nativeElement, this.type, this.io);
    }
};
SocketDirective.ɵfac = function SocketDirective_Factory(t) { return new (t || SocketDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NodeService)); };
SocketDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: SocketDirective, selectors: [["", "rete-socket", ""]], inputs: { io: "io" } });
SocketDirective.ctorParameters = () => [
    { type: ElementRef },
    { type: NodeService }
];
tslib_1.__decorate([
    Input()
], SocketDirective.prototype, "io", void 0);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SocketDirective, [{
        type: Directive,
        args: [{
                selector: '[rete-socket]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.NodeService }]; }, { io: [{
            type: Input
        }] }); })();
export { SocketDirective };
