// rete-angular-render-plugin/esm2015/node/node.component.js
import * as tslib_1 from "tslib";
import { Component, Input, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
import { NodeService } from '../node.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '../node.service';
import * as ɵngcc2 from '@angular/common';
import * as ɵngcc3 from '../socket/socket.component';
import * as ɵngcc4 from '../socket.directive';
import * as ɵngcc5 from '../control.directive';
import * as ɵngcc6 from '../kebab.pipe';

function NodeComponent_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 5);
    ɵngcc0.ɵɵelementStart(1, "div", 6);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(3, "rete-socket", 7);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const output_r3 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(output_r3.name);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("io", output_r3)("socket", output_r3.socket);
} }
function NodeComponent_div_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 8);
} if (rf & 2) {
    const control_r4 = ctx.$implicit;
    ɵngcc0.ɵɵproperty("rete-control", control_r4);
} }
function NodeComponent_div_6_div_2_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 12);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const input_r5 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(input_r5.name);
} }
function NodeComponent_div_6_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 13);
} if (rf & 2) {
    const input_r5 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵproperty("rete-control", input_r5.control);
} }
function NodeComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵelement(1, "rete-socket", 7);
    ɵngcc0.ɵɵtemplate(2, NodeComponent_div_6_div_2_Template, 2, 1, "div", 10);
    ɵngcc0.ɵɵtemplate(3, NodeComponent_div_6_div_3_Template, 1, 1, "div", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const input_r5 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("io", input_r5)("socket", input_r5.socket);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", !input_r5.showControl());
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", input_r5.showControl());
} }
const _c0 = function (a0, a1) { return [a0, a1]; };
let NodeComponent = class NodeComponent {
    constructor(service, cdr) {
        this.service = service;
        this.cdr = cdr;
    }
    ngOnInit() {
        this.service.setBindings(this.bindSocket, this.bindControl);
        this.node.update = () => this.cdr.detectChanges();
    }
    get inputs() {
        return Array.from(this.node.inputs.values());
    }
    get outputs() {
        return Array.from(this.node.outputs.values());
    }
    get controls() {
        return Array.from(this.node.controls.values());
    }
    selected() {
        return this.editor.selected.contains(this.node) ? 'selected' : '';
    }
};
NodeComponent.ɵfac = function NodeComponent_Factory(t) { return new (t || NodeComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NodeService), ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ChangeDetectorRef)); };
NodeComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NodeComponent, selectors: [["ng-component"]], inputs: { editor: "editor", node: "node", bindSocket: "bindSocket", bindControl: "bindControl" }, features: [ɵngcc0.ɵɵProvidersFeature([NodeService])], decls: 7, vars: 10, consts: [[1, "node", 3, "ngClass"], [1, "title"], ["class", "output", 4, "ngFor", "ngForOf"], ["class", "control", 3, "rete-control", 4, "ngFor", "ngForOf"], ["class", "input", 4, "ngFor", "ngForOf"], [1, "output"], [1, "output-title"], ["rete-socket", "", 3, "io", "socket"], [1, "control", 3, "rete-control"], [1, "input"], ["class", "input-title", 4, "ngIf"], ["class", "input-control", 3, "rete-control", 4, "ngIf"], [1, "input-title"], [1, "input-control", 3, "rete-control"]], template: function NodeComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵpipe(1, "kebab");
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵtext(3);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(4, NodeComponent_div_4_Template, 4, 3, "div", 2);
        ɵngcc0.ɵɵtemplate(5, NodeComponent_div_5_Template, 1, 1, "div", 3);
        ɵngcc0.ɵɵtemplate(6, NodeComponent_div_6_Template, 4, 4, "div", 4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpipeBind1(1, 5, ɵngcc0.ɵɵpureFunction2(7, _c0, ctx.selected(), ctx.node.name)));
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵtextInterpolate(ctx.node.name);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.outputs);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.controls);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.inputs);
    } }, directives: [ɵngcc2.NgClass, ɵngcc2.NgForOf, ɵngcc3.SocketComponent, ɵngcc4.SocketDirective, ɵngcc5.ControlDirective, ɵngcc2.NgIf], pipes: [ɵngcc6.KebabPipe], styles: [".node[_ngcontent-%COMP%]{background:rgba(110,136,255,.8);border:2px solid #4e58bf;border-radius:10px;cursor:pointer;min-width:180px;height:auto;padding-bottom:6px;box-sizing:content-box;position:relative;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.node[_ngcontent-%COMP%]:hover{background:rgba(130,153,255,.8)}.node.selected[_ngcontent-%COMP%]{background:#ffd92c;border-color:#e3c000}.node[_ngcontent-%COMP%]   .title[_ngcontent-%COMP%]{color:#fff;font-family:sans-serif;font-size:18px;padding:8px}.node[_ngcontent-%COMP%]   .output[_ngcontent-%COMP%]{text-align:right}.node[_ngcontent-%COMP%]   .input[_ngcontent-%COMP%]{text-align:left}.node[_ngcontent-%COMP%]   .input-title[_ngcontent-%COMP%], .node[_ngcontent-%COMP%]   .output-title[_ngcontent-%COMP%]{vertical-align:middle;color:#fff;display:inline-block;font-family:sans-serif;font-size:14px;margin:6px;line-height:24px}.node[_ngcontent-%COMP%]   .input-title[hidden][_ngcontent-%COMP%], .node[_ngcontent-%COMP%]   .output-title[hidden][_ngcontent-%COMP%]{display:none}.node[_ngcontent-%COMP%]   .input-control[_ngcontent-%COMP%]{z-index:1;width:calc(100% - 36px);vertical-align:middle;display:inline-block}.node[_ngcontent-%COMP%]   .control[_ngcontent-%COMP%]{padding:6px 18px}"], changeDetection: 0 });
NodeComponent.ctorParameters = () => [
    { type: NodeService },
    { type: ChangeDetectorRef }
];
tslib_1.__decorate([
    Input()
], NodeComponent.prototype, "editor", void 0);
tslib_1.__decorate([
    Input()
], NodeComponent.prototype, "node", void 0);
tslib_1.__decorate([
    Input()
], NodeComponent.prototype, "bindSocket", void 0);
tslib_1.__decorate([
    Input()
], NodeComponent.prototype, "bindControl", void 0);
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NodeComponent, [{
        type: Component,
        args: [{
                template: "<div class=\"node\" [ngClass]=\"[selected(), node.name] | kebab\">\n    <div class=\"title\">{{node.name}}</div>\n    <div class=\"output\" *ngFor=\"let output of outputs\">\n        <div class=\"output-title\">{{output.name}}</div>\n        <rete-socket rete-socket [io]=\"output\" [socket]=\"output.socket\"></rete-socket>\n    </div>\n    <div class=\"control\" *ngFor=\"let control of controls\" [rete-control]=\"control\"></div>\n    <div class=\"input\" *ngFor=\"let input of inputs\">\n        <rete-socket rete-socket [io]=\"input\" [socket]=\"input.socket\"></rete-socket>\n        <div class=\"input-title\" *ngIf=\"!input.showControl()\">{{input.name}}</div>\n        <div class=\"input-control\" *ngIf=\"input.showControl()\" [rete-control]=\"input.control\"></div>\n    </div>\n</div>",
                providers: [NodeService],
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [".node{background:rgba(110,136,255,.8);border:2px solid #4e58bf;border-radius:10px;cursor:pointer;min-width:180px;height:auto;padding-bottom:6px;box-sizing:content-box;position:relative;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.node:hover{background:rgba(130,153,255,.8)}.node.selected{background:#ffd92c;border-color:#e3c000}.node .title{color:#fff;font-family:sans-serif;font-size:18px;padding:8px}.node .output{text-align:right}.node .input{text-align:left}.node .input-title,.node .output-title{vertical-align:middle;color:#fff;display:inline-block;font-family:sans-serif;font-size:14px;margin:6px;line-height:24px}.node .input-title[hidden],.node .output-title[hidden]{display:none}.node .input-control{z-index:1;width:calc(100% - 36px);vertical-align:middle;display:inline-block}.node .control{padding:6px 18px}"]
            }]
    }], function () { return [{ type: ɵngcc1.NodeService }, { type: ɵngcc0.ChangeDetectorRef }]; }, { editor: [{
            type: Input
        }], node: [{
            type: Input
        }], bindSocket: [{
            type: Input
        }], bindControl: [{
            type: Input
        }] }); })();
export { NodeComponent };
