// rete-angular-render-plugin/esm5/socket/socket.component.js
import * as tslib_1 from "tslib";
import { Component, Input, ChangeDetectionStrategy } from '@angular/core';
import { Input as ReteInput } from 'rete';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';

var _c0 = function (a0, a1) { return [a0, a1]; };
function SocketComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 1);
} if (rf & 2) {
    var ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction2(2, _c0, ctx_r0.type, ctx_r0.socket.name))("title", ctx_r0.socket.name);
} }
var SocketComponent = /** @class */ (function () {
    function SocketComponent() {
    }
    Object.defineProperty(SocketComponent.prototype, "type", {
        get: function () {
            return this.io instanceof ReteInput ? 'input' : 'output';
        },
        enumerable: true,
        configurable: true
    });
    tslib_1.__decorate([
        Input()
    ], SocketComponent.prototype, "socket", void 0);
    tslib_1.__decorate([
        Input()
    ], SocketComponent.prototype, "io", void 0);
SocketComponent.ɵfac = function SocketComponent_Factory(t) { return new (t || SocketComponent)(); };
SocketComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: SocketComponent, selectors: [["rete-socket"]], inputs: { socket: "socket", io: "io" }, decls: 1, vars: 1, consts: [["class", "socket", 3, "ngClass", "title", 4, "ngIf"], [1, "socket", 3, "ngClass", "title"]], template: function SocketComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵtemplate(0, SocketComponent_div_0_Template, 1, 5, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.socket);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgClass], styles: ["[_nghost-%COMP%]{display:inline-block}.socket[_ngcontent-%COMP%]{display:inline-block;cursor:pointer;border:1px solid #fff;border-radius:12px;width:24px;height:24px;margin:6px;vertical-align:middle;background:#96b38a;z-index:2;box-sizing:border-box}.socket[_ngcontent-%COMP%]:hover{border-width:4px}.socket.multiple[_ngcontent-%COMP%]{border-color:#ff0}.socket.output[_ngcontent-%COMP%]{margin-right:-12px}.socket.input[_ngcontent-%COMP%]{margin-left:-12px}"], changeDetection: 0 });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(SocketComponent, [{
        type: Component,
        args: [{
                selector: 'rete-socket',
                template: "<div *ngIf=\"socket\" class=\"socket\" [ngClass]=\"[type, socket.name]\" [title]=\"socket.name\"></div>",
                changeDetection: ChangeDetectionStrategy.OnPush,
                styles: [":host{display:inline-block}.socket{display:inline-block;cursor:pointer;border:1px solid #fff;border-radius:12px;width:24px;height:24px;margin:6px;vertical-align:middle;background:#96b38a;z-index:2;box-sizing:border-box}.socket:hover{border-width:4px}.socket.multiple{border-color:#ff0}.socket.output{margin-right:-12px}.socket.input{margin-left:-12px}"]
            }]
    }], function () { return []; }, { socket: [{
            type: Input
        }], io: [{
            type: Input
        }] }); })();
    return SocketComponent;
}());
export { SocketComponent };
