// @materia-ui/ngx-monaco-editor/esm5/lib/monaco-editor.module.js
/**
 * @fileoverview added by tsickle
 * Generated from: lib/monaco-editor.module.ts
 * @suppress {checkTypes,constantProperty,extraRequire,missingOverride,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import { NgModule } from '@angular/core';
import { MonacoEditorLoaderDirective } from './directives/monaco-editor-loader.directive';
import { MonacoEditorComponent } from './components/monaco-editor/monaco-editor.component';
import { ResizedDirective } from './directives/resized-event.directive';
import { MonacoDiffEditorComponent } from './components/monaco-diff-editor/monaco-diff-editor.component';
import * as ɵngcc0 from '@angular/core';
var MonacoEditorModule = /** @class */ (function () {
    function MonacoEditorModule() {
    }
MonacoEditorModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: MonacoEditorModule });
MonacoEditorModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function MonacoEditorModule_Factory(t) { return new (t || MonacoEditorModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(MonacoEditorModule, { declarations: function () { return [MonacoEditorLoaderDirective,
        MonacoEditorComponent,
        ResizedDirective,
        MonacoDiffEditorComponent]; }, exports: function () { return [MonacoEditorLoaderDirective,
        MonacoEditorComponent,
        MonacoDiffEditorComponent,
        ResizedDirective]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(MonacoEditorModule, [{
        type: NgModule,
        args: [{
                imports: [],
                declarations: [
                    MonacoEditorLoaderDirective,
                    MonacoEditorComponent,
                    ResizedDirective,
                    MonacoDiffEditorComponent
                ],
                exports: [
                    MonacoEditorLoaderDirective,
                    MonacoEditorComponent,
                    MonacoDiffEditorComponent,
                    ResizedDirective
                ],
                entryComponents: [MonacoEditorComponent, MonacoDiffEditorComponent]
            }]
    }], function () { return []; }, null); })();
    return MonacoEditorModule;
}());
export { MonacoEditorModule };
