// rete-angular-render-plugin/esm5/control.directive.js
import * as tslib_1 from "tslib";
import { Directive, Input, ElementRef, OnInit } from '@angular/core';
import { NodeService } from './node.service';
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from './node.service';
var ControlDirective = /** @class */ (function () {
    function ControlDirective(el, service) {
        this.el = el;
        this.service = service;
    }
    ControlDirective.prototype.ngOnInit = function () {
        this.service.bindControl(this.el.nativeElement, this.control);
    };
    ControlDirective.ctorParameters = function () { return [
        { type: ElementRef },
        { type: NodeService }
    ]; };
    tslib_1.__decorate([
        Input('rete-control')
    ], ControlDirective.prototype, "control", void 0);
ControlDirective.ɵfac = function ControlDirective_Factory(t) { return new (t || ControlDirective)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef), ɵngcc0.ɵɵdirectiveInject(ɵngcc1.NodeService)); };
ControlDirective.ɵdir = ɵngcc0.ɵɵdefineDirective({ type: ControlDirective, selectors: [["", "rete-control", ""]], inputs: { control: ["rete-control", "control"] } });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ControlDirective, [{
        type: Directive,
        args: [{
                selector: '[rete-control]'
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }, { type: ɵngcc1.NodeService }]; }, { control: [{
            type: Input,
            args: ['rete-control']
        }] }); })();
    return ControlDirective;
}());
export { ControlDirective };
