// rete-angular-render-plugin/esm5/index.js
import { NodeComponent } from './node/node.component';
function install(editor, params) {
    if (params === void 0) { params = {}; }
    editor.on('rendernode', function (_a) {
        var el = _a.el, node = _a.node, component = _a.component, bindControl = _a.bindControl, bindSocket = _a.bindSocket;
        var ngComponent = component;
        if (ngComponent.render && ngComponent.render !== 'angular')
            return;
        var element = document.createElement('rete-element');
        var props = element;
        props.component = ngComponent.component || params.component || NodeComponent;
        props.props = Object.assign({}, ngComponent.props || {}, {
            node: node,
            editor: editor,
            bindControl: bindControl,
            bindSocket: bindSocket
        });
        el.appendChild(element);
    });
    editor.on('rendercontrol', function (_a) {
        var el = _a.el, control = _a.control;
        var ngControl = control;
        if (ngControl.render && ngControl.render !== 'angular')
            return;
        var element = document.createElement('rete-element');
        var props = element;
        props.component = ngControl.component;
        props.props = ngControl.props;
        el.appendChild(element);
    });
    editor.on(['connectioncreated', 'connectionremoved'], function (connection) {
        connection.output.node.update();
        connection.input.node.update();
    });
    editor.on('nodeselected', function () {
        editor.nodes.forEach(function (n) { return n.update(); });
    });
}
export var AngularRenderPlugin = {
    name: 'angular-render',
    install: install
};
export { ReteModule } from './module';
export { NodeService } from './node.service';
export { NodeComponent } from './node/node.component';
export { SocketComponent } from './socket/socket.component';
