// rete-angular-render-plugin/esm5/module.js
import * as tslib_1 from "tslib";
import { NgModule, Injector } from '@angular/core';
import { CommonModule } from '@angular/common';
import { createCustomElement } from '@angular/elements';
import { KebabPipe } from './kebab.pipe';
import { NodeComponent } from './node/node.component';
import { SocketComponent } from './socket/socket.component';
import { ControlDirective } from './control.directive';
import { SocketDirective } from './socket.directive';
import { CustomComponent } from './custom.component';
import * as ɵngcc0 from '@angular/core';
var ReteModule = /** @class */ (function () {
    function ReteModule(injector) {
        var CustomElement = createCustomElement(CustomComponent, { injector: injector });
        if (!customElements.get('rete-element'))
            customElements.define('rete-element', CustomElement);
    }
    ReteModule.ctorParameters = function () { return [
        { type: Injector }
    ]; };
ReteModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ReteModule });
ReteModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ReteModule_Factory(t) { return new (t || ReteModule)(ɵngcc0.ɵɵinject(ɵngcc0.Injector)); }, providers: [
        KebabPipe,
        ControlDirective
    ], imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ReteModule, { declarations: function () { return [NodeComponent,
        SocketComponent,
        CustomComponent,
        ControlDirective,
        SocketDirective,
        KebabPipe]; }, imports: function () { return [CommonModule]; }, exports: function () { return [NodeComponent,
        CustomComponent,
        SocketComponent,
        ControlDirective,
        SocketDirective,
        KebabPipe]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ReteModule, [{
        type: NgModule,
        args: [{
                declarations: [
                    NodeComponent,
                    SocketComponent,
                    CustomComponent,
                    ControlDirective,
                    SocketDirective,
                    KebabPipe,
                ],
                imports: [
                    CommonModule
                ],
                providers: [
                    KebabPipe,
                    ControlDirective
                ],
                exports: [
                    NodeComponent,
                    CustomComponent,
                    SocketComponent,
                    ControlDirective,
                    SocketDirective,
                    KebabPipe
                ],
                entryComponents: [
                    NodeComponent,
                    SocketComponent,
                    CustomComponent
                ]
            }]
    }], function () { return [{ type: ɵngcc0.Injector }]; }, null); })();
    return ReteModule;
}());
export { ReteModule };
