// rete-angular-render-plugin/esm5/module.js
import * as tslib_1 from "tslib";
import { NgModule, Injector } from '@angular/core';
import { CommonModule } from '@angular/common';
import { createCustomElement } from '@angular/elements';
import { KebabPipe } from './kebab.pipe';
import { NodeComponent } from './node/node.component';
import { SocketComponent } from './socket/socket.component';
import { ControlDirective } from './control.directive';
import { SocketDirective } from './socket.directive';
import { CustomComponent } from './custom.component';
var ReteModule = /** @class */ (function () {
    function ReteModule(injector) {
        var CustomElement = createCustomElement(CustomComponent, { injector: injector });
        if (!customElements.get('rete-element'))
            customElements.define('rete-element', CustomElement);
    }
    ReteModule.ctorParameters = function () { return [
        { type: Injector }
    ]; };
    ReteModule = tslib_1.__decorate([
        NgModule({
            declarations: [
                NodeComponent,
                SocketComponent,
                CustomComponent,
                ControlDirective,
                SocketDirective,
                KebabPipe,
            ],
            imports: [
                CommonModule
            ],
            providers: [
                KebabPipe,
                ControlDirective
            ],
            exports: [
                NodeComponent,
                CustomComponent,
                SocketComponent,
                ControlDirective,
                SocketDirective,
                KebabPipe
            ],
            entryComponents: [
                NodeComponent,
                SocketComponent,
                CustomComponent
            ]
        })
    ], ReteModule);
    return ReteModule;
}());
export { ReteModule };
