// rete-angular-render-plugin/esm5/node/node.component.js
import * as tslib_1 from "tslib";
import { Component, Input, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
import { NodeService } from '../node.service';
var NodeComponent = /** @class */ (function () {
    function NodeComponent(service, cdr) {
        this.service = service;
        this.cdr = cdr;
    }
    NodeComponent.prototype.ngOnInit = function () {
        var _this = this;
        this.service.setBindings(this.bindSocket, this.bindControl);
        this.node.update = function () { return _this.cdr.detectChanges(); };
    };
    Object.defineProperty(NodeComponent.prototype, "inputs", {
        get: function () {
            return Array.from(this.node.inputs.values());
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NodeComponent.prototype, "outputs", {
        get: function () {
            return Array.from(this.node.outputs.values());
        },
        enumerable: true,
        configurable: true
    });
    Object.defineProperty(NodeComponent.prototype, "controls", {
        get: function () {
            return Array.from(this.node.controls.values());
        },
        enumerable: true,
        configurable: true
    });
    NodeComponent.prototype.selected = function () {
        return this.editor.selected.contains(this.node) ? 'selected' : '';
    };
    NodeComponent.ctorParameters = function () { return [
        { type: NodeService },
        { type: ChangeDetectorRef }
    ]; };
    tslib_1.__decorate([
        Input()
    ], NodeComponent.prototype, "editor", void 0);
    tslib_1.__decorate([
        Input()
    ], NodeComponent.prototype, "node", void 0);
    tslib_1.__decorate([
        Input()
    ], NodeComponent.prototype, "bindSocket", void 0);
    tslib_1.__decorate([
        Input()
    ], NodeComponent.prototype, "bindControl", void 0);
    NodeComponent = tslib_1.__decorate([
        Component({
            template: "<div class=\"node\" [ngClass]=\"[selected(), node.name] | kebab\">\n    <div class=\"title\">{{node.name}}</div>\n    <div class=\"output\" *ngFor=\"let output of outputs\">\n        <div class=\"output-title\">{{output.name}}</div>\n        <rete-socket rete-socket [io]=\"output\" [socket]=\"output.socket\"></rete-socket>\n    </div>\n    <div class=\"control\" *ngFor=\"let control of controls\" [rete-control]=\"control\"></div>\n    <div class=\"input\" *ngFor=\"let input of inputs\">\n        <rete-socket rete-socket [io]=\"input\" [socket]=\"input.socket\"></rete-socket>\n        <div class=\"input-title\" *ngIf=\"!input.showControl()\">{{input.name}}</div>\n        <div class=\"input-control\" *ngIf=\"input.showControl()\" [rete-control]=\"input.control\"></div>\n    </div>\n</div>",
            providers: [NodeService],
            changeDetection: ChangeDetectionStrategy.OnPush,
            styles: [".node{background:rgba(110,136,255,.8);border:2px solid #4e58bf;border-radius:10px;cursor:pointer;min-width:180px;height:auto;padding-bottom:6px;box-sizing:content-box;position:relative;-webkit-user-select:none;-moz-user-select:none;-ms-user-select:none;user-select:none}.node:hover{background:rgba(130,153,255,.8)}.node.selected{background:#ffd92c;border-color:#e3c000}.node .title{color:#fff;font-family:sans-serif;font-size:18px;padding:8px}.node .output{text-align:right}.node .input{text-align:left}.node .input-title,.node .output-title{vertical-align:middle;color:#fff;display:inline-block;font-family:sans-serif;font-size:14px;margin:6px;line-height:24px}.node .input-title[hidden],.node .output-title[hidden]{display:none}.node .input-control{z-index:1;width:calc(100% - 36px);vertical-align:middle;display:inline-block}.node .control{padding:6px 18px}"]
        })
    ], NodeComponent);
    return NodeComponent;
}());
export { NodeComponent };
